rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -2560,53 +2803,48 @@ extern void ppu_precompile(std::vector<std::string>& dir_queue, std::vector<lv2_
 
 			std::string upper = fmt::to_upper(entry.name);
 
-			// Check .sprx filename
-			if (upper.ends_with(".SPRX") && entry.name != "libfs_utility_init.sprx"sv)
+			// Skip already loaded modules or HLEd ones
```

Comment:
This is another improvement unrelated to the saving of states. You should have opened incremental PRs with the supporting fixes for easier risk management.

Reply:
No, it's related to savestates. It's required to skip precompilation of already loaded modules with the addition of OVL modules checking which comes with savestaing MGS4.

Reply:
It is standalone though. That's what I mean. I understand that it is needed for savestates to work, but it doesn't contain the core saving of states (serialization/deserialization) and would have been good in a part 1 PR. Again, I don't expect anything to change, but I have to give advice anyway for the sake of future PRs.